src/stdlib_math_arange.fypp

```diff
@@ -0,0 +1,56 @@
+#:include "common.fypp"
+submodule(stdlib_math) stdlib_math_arange
+
+    implicit none
+
+contains
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    !> `arange` creates a vector of the `${t1}$` type 
+    !>  with evenly spaced values within a given interval.
+    pure module function arange_${t1[0]}$_${k1}$(start, end, by) result(result)
+
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in), optional :: end, by
+        ${t1}$, allocatable :: result(:)
+        
+        ${t1}$ :: start_, end_, by_
+        integer :: i
+
+        start_ = merge(start, 1.0_${k1}$, present(end))
+        end_   = merge(end,   start,      present(end))
+        by_    = sign(merge(merge(by, 1.0_${k1}$, by /= 0.0_${k1}$), &
+                 1.0_${k1}$, present(by)), end_ - start_)
+
+        allocate(result(floor((end_ - start_)/by_) + 1))
```

Comment:
```fortran
by /= 0.0_${k1}$
```
When the `by` value is a very small amount, here becomes dangerous. Do we need to set a limit for it?